### PR TITLE
SILA-2324: Built User Retrieve data and integrated Registration data accordion and table components etc.

### PR DESCRIPTION
**Pull request by ArcGate-Vinod**
- Integrated Registration data UI.
- Built User Retrieve data and integrated Registration data accordion and table components etc.

Thanks!

### CODE REVIEW

File: src/components/register/InstantAchKYCForm.js [hunk 1]
```diff
@@ -4,62 +4,65 @@ import NumberFormat from 'react-number-format';
...
-const InstantAchKYCForm = ({ errors, children }) => {
+const InstantAchKYCForm = ({ errors, isHide, app, children }) => {
   const [receiveSMS, setReceiveSMS] = useState(false);
   const onSMSChange = (e) => {
     setReceiveSMS(e.target.checked);
   }
   
   return (
-    <>
+    <div className={isHide ? 'd-none' : ''}>
```
[the remark this answers is not in the document]
Comment (ArcGate-Vinod) [reply src/components/register/InstantAchKYCForm.js]: Hi @amack300,
Change has been done.

Thanks!

File: src/components/register/InstantAchKYCForm.js [hunk 2]
```diff
@@ -4,62 +4,65 @@ import NumberFormat from 'react-number-format';
...
+      <p className="text-muted">This KYC Level requires us to gather more information from you. Please fill out the required fields below.</p>
+
+      <p className="text-right text-lg text-warning">All fields are required for this KYC level.</p>
       <Form.Row>
         <Form.Group as={Col} md="6" controlId="registerFirstName" className="required">
-          <Form.Control required placeholder="First Name" name="firstName" />
+          <Form.Control required placeholder="First Name" name="firstName" defaultValue={(app.activeUser) ? app.activeUser.firstName : ''} />
```
[the remark this answers is not in the document]
Comment (ArcGate-Vinod) [reply src/components/register/InstantAchKYCForm.js]: Hi @amack300,
Changes have been done.

Thanks!

File: src/components/register/KYCLiteForm.js [hunk 3]
```diff
@@ -2,39 +2,40 @@ import React from 'react';
 import { Form, Col } from 'react-bootstrap';
 import NumberFormat from 'react-number-format';
 
-const KYCLiteForm = ({ errors, children }) => {
+const KYCLiteForm = ({ errors, isHide, app, children }) => {
   
   return (
-    <>
+    <div className={isHide ? 'd-none' : ''}>
```
[the remark this answers is not in the document]
Comment (ArcGate-Vinod) [reply src/components/register/KYCLiteForm.js]: Hi @amack300,
Change has been done.

Thanks!

File: src/components/register/ReceiveOnlyKYCForm.js [hunk 4]
```diff
@@ -1,23 +1,24 @@
 import React from 'react';
 import { Form, Col } from 'react-bootstrap';
 
-const ReceiveOnlyKYCForm = ({ errors, children }) => {
+const ReceiveOnlyKYCForm = ({ errors, isHide, app, children }) => {
   
   return (
-    <>
+    <div className={isHide ? 'd-none' : ''}>
```
[the remark this answers is not in the document]
Comment (ArcGate-Vinod) [reply src/components/register/ReceiveOnlyKYCForm.js]: Hi @amack300,
Change has been done.

Thanks!

File: src/components/register/RegisterUserForm.js [hunk 5]
```diff
@@ -3,106 +3,346 @@ import { Form, Button } from 'react-bootstrap';
...
+
+      if(preferredKyc !== RECEIVE_ONLY_KYC) {
+        validationErrors = { ...validationErrors, contact: {} }
+        if (e.target.email.value !== app.activeUser.email) {
+          try {
+            const emailUpdateData = {};
+            emailUpdateData.email = e.target.email ? e.target.email.value : '';
+            emailUpdateData.uuid = entityRes.data.emails[0] ? entityRes.data.emails[0]['uuid'] : '';
+
+            let ApiEndpoint = '/add/email';
```
[the remark this answers is not in the document]
Comment (ArcGate-Vinod) [reply src/components/register/RegisterUserForm.js]: Hi @amack300,
Changes have been done.

Thanks!

File: src/assets/styles/_accordion.scss [hunk 6]
```diff
@@ -0,0 +1,42 @@
+$no-repeat:no-repeat;
+$borderNone: none;
+$backgroundTransparent: transparent;
+.accordion > .card:not(:last-of-type){
+  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
+}
+.btn-save{
```
[the remark this answers is not in the document]
Comment (ArcGate-Vinod) [reply src/assets/styles/_accordion.scss]: Hi @amack300,
Change has been done.

Thanks!

File: src/assets/styles/_accordion.scss [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    .accordion-icon{
+      background-image: url(../images/accordion_minus.svg);
+      background-repeat:$no-repeat;
+      height: 4px;
+      width: 26px;
+      border: $borderNone;
+      background-color: $backgroundTransparent;
+    }
+  }
+  .form-control{
```
[the remark this answers is not in the document]
Comment (ArcGate-Vinod) [reply src/assets/styles/_accordion.scss]: Hi @amack300,
Changes have been done.

Thanks!

File: src/assets/styles/views/_registeruser.scss [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+//SMS Notifications
+.sms-notifications{
+  margin-top: 10px;
+  font-weight: 500;
+  font-size: 1.125rem;
+  line-height: 22px;
+  letter-spacing: 0.03em;
+  span{
+    color: $primary;
```
[the remark this answers is not in the document]
Comment (ArcGate-Vinod) [reply src/assets/styles/views/_registeruser.scss]: Hi @amack300,
Change has been done.

Thanks!

File: src/components/register/DefaultKYCForm.js [hunk 9]
```diff
@@ -4,64 +4,67 @@ import NumberFormat from 'react-number-format';
 
 import { STATES_ARRAY } from '../../constants';
 
-const DefaultKYCForm = ({ errors, children }) => {
+const DefaultKYCForm = ({ errors, isHide, app, children }) => {
   
   return (
-    <>
+    <div className={isHide ? 'd-none' : ''}>
```
[the remark this answers is not in the document]
Comment (ArcGate-Vinod) [reply src/components/register/DefaultKYCForm.js]: Hi @amack300,
Change has been done.

Thanks!

File: src/components/register/RegisterDataForm.js [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+        <Table responsive hover>
+          <thead>
+            <tr>
+              <th className="font-weight-bold">Registration Field</th>
+              <th className="font-weight-bold">Data</th>
+              <th className="font-weight-bold">Action</th>
+            </tr>
+          </thead>
+          <tbody>
+            {KYC_REGISTER_FIELDS_ARRAY.map((fieldsOption, index) => {
```
[the remark this answers is not in the document]
Comment (ArcGate-Vinod) [reply src/components/register/RegisterDataForm.js]: Hi @amack300,
Changes have been done.

Thanks!

**Review by amack300 — APPROVED** [review]
LGTM

**Pull request merged**